libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -174,13 +175,11 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
                         request(callback, modifiedRerouteOption)
                     }
                     is RouteOptionsUpdater.RouteOptionsResult.Error -> {
-                        mainJobController.scope.launch {
```

Comment:
Is there a reason why this was not synchronous? Maybe we didn't want to notify observers synchronously because we supposed they can do some heavy work that would prevent us from executing next lines of our synchronous code?

Reply:
We already do update the state synchronously above anyway. @RingerJK do you remember the reason?

Reply:
Right...

Reply:
what might happen is `reroute()` can be invoked in non-main thread, but we spread updates in the main thread only. That's why it was added originally. Now we have `@UiThread` annotations, so it's ok to remove the line

Reply:
Thanks @RingerJK, we should be good then.